### PR TITLE
No more '..' in the file prompt (#2104)

### PR DESCRIPTION
**Pull request by jso8910**
Another thing in #2104

### CODE REVIEW

**Comment by jso8910** [comment]
Plus with this the /usr/bin issue with tabbing mentioned in the above issue can no longer occurred.

File: tests/unit/mainwindow/test_prompt.py [hunk 1]
```diff
@@ -48,9 +48,8 @@ def _get_prompt_func(path):
         return _get_prompt_func
 
     @pytest.mark.parametrize('steps, where, subfolder', [
-        (1, 'next', '..'),
-        (1, 'prev', 'c'),
```
Comment (lufte) [inline tests/unit/mainwindow/test_prompt.py:52]: You can leave this one as it still should work. Going to the previous item once still falls in `c`.

**Review by lufte — APPROVED** [review]
Thanks for this. I've always found it annoying but never realized there was a ticket for it.

I left a comment on a small detail on the tests, but besides that it's :+1: from me.

File: tests/unit/mainwindow/test_prompt.py (hunk 1 above)
Comment (jso8910) [reply tests/unit/mainwindow/test_prompt.py:52]: Are you saying line 52 still works or line 51.

Comment (lufte) [reply tests/unit/mainwindow/test_prompt.py:52]: Line 52. Going `1` step in the `prev` direction should still leave us in the `c` folder.

**Comment by The-Compiler** [comment]
Thanks @jso8910 and @lufte, and sorry for the delay! I readded the test @lufte commented on:

- 621087032 Restore deleted test

One thing that kept me back from merging this was that I still wanted an easy and discoverable way for people to go up a directory - if you don't know that typing `../` works, you might not think of doing that. While we have a `:rl-unix-filename-rubout` command, it had some issues (#4561, and also using `/` even on Windows). I thus ended up implementing a new `:rl-filename-rubout` without those issues, added a default binding for it, and made sure it gets shown in the list of commands for filename prompts:

image

Along the way, I also found a bug in the readline commands, more about that over in https://github.com/qutebrowser/qutebrowser/issues/4561#issuecomment-1084983084.

- ebdd791b7 readline: Allow passing args to cmdutils.register
- ab65c542a Add :rl-rubout and :rl-filename-rubout
- 1af602b25 readline: Properly delete first character
- ecf63880d Fix lint / changelog

With that all implemented/fixed, I hope we now have a suitable alternative without the common annoyance of tabbing to `..` when going through the list!

**Comment by JingMatrix** [comment]
I have a different idea then this fix, in fact, in my local version, I changed this line

https://github.com/qutebrowser/qutebrowser/blob/01a402dcfc76c24d872a6d6021decc4c1542e197/qutebrowser/mainwindow/prompt.py#L655

by just remove the second part, as my fork :

https://github.com/JingMatrix/qutebrowser/blob/81efff6af49139bffa3348f251458990b2c60e0d/qutebrowser/mainwindow/prompt.py#L655

It solved the problem and I think this solution is more direct without bothering for a new shortcut.

**Comment by The-Compiler** [comment]
@JingMatrix No point in displaying `..` if there's no way to select it using the keyboard - also, too late for that now, I'd rather not change things yet again.

**Pull request merged**